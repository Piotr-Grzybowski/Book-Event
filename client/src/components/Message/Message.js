import React from 'react';
import './Message.css';

function Message({ message, loader, errors }) {
  const errorsList = errors.map((error, index) =>
    <li key={index}>{error.msg}</li>
  )
  if (message && loader.active) {
    return(
      <div className="loader-wrapper"><div className="loader"  data-testid="loader"></div></div>
    )
  } else if (message) {
    return (
      <div id="myModal" className="modal">
        <div className="modal-content">
          <div className="modal-header" data-testid="modal-header">
            <span className="close">&times;</span>
            <h2>{loader.error ? 'Failure' : 'Success'}</h2>
          </div>
          <div className="modal-body" data-testid="message">
            <p>{message}</p>
            {errors.length ? <div><h3>Errors: </h3><ul data-testid="errors">{errorsList}</ul></div> : ''}
          </div>
          <div className="modal-footer" data-testid="modal-footer">
            <h3>{loader.error ? 'Something went wrong!' : 'Everything went smoothly!'}</h3>
          </div>
        </div>
      </div>
    )
  }
  return (
    <div className="emptyDiv" data-testid="emptyDiv"></div>
  )
}

Message.displayName = 'Message';

export default Message;