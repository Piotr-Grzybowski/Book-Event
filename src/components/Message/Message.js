import React from 'react';
import './Message.css';

function Message({ message, loader }) {
  if (message) {
    if (!loader.active) {
      return (
        <div className="message" data-testid="message">
          <p>{message}</p>
        </div>
      )
    }
    return (
      <div className='loader' data-testid="loader">
      </div>
    )
  }
  return (
    <div className="emptyDiv" data-testid="emptyDiv"></div>
  )
}

export default Message;