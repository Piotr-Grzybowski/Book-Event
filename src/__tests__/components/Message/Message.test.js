import { cleanup, getNodeText } from '@testing-library/react';
import React from 'react';
import Message from '../../../components/Message/Message';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import register from 'ignore-styles';
register(['.css', '.sass', '.scss']);


describe('Message component', () => {
  const message = 'some example message';

  afterEach(cleanup);

  it('should match snapshot', () => {
    const snapshot = renderer.create(<Message message={message} loader={{active: false}} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('should render message when message is truthy and loader.active is false', () => {
    const { container } = render(<Message message={message} loader={{active: false}} />);
    const messageWrapper = container.querySelector('[data-testid="message"]');
    const emptyDivWraper = container.querySelector('[data-testid="emptyDiv"]');
    expect(getNodeText(messageWrapper.firstChild)).toMatch(message);
    expect(emptyDivWraper).not.toBeInTheDocument();
  });
  it('should render empty div when message is falsy', () => {
    const { container } = render(<Message message='' loader={{active: false}} />);
    const messageWrapper = container.querySelector('[data-testid="message"]');
    const loaderWrapper = container.querySelector('[data-testid="loader"]');
    const emptyDivWraper = container.querySelector('[data-testid="emptyDiv"]');
    expect(messageWrapper).not.toBeInTheDocument();
    expect(loaderWrapper).not.toBeInTheDocument();
    expect(emptyDivWraper).toBeInTheDocument();
    expect(emptyDivWraper.firstChild).not.toBeInTheDocument();
  })
  it('should render loader when message is truthy and loader.active is true', () => {
    const { container } = render(<Message message={message} loader={{active: true}} />);
    const messageWrapper = container.querySelector('[data-testid="message"]');
    const loaderWrapper = container.querySelector('[data-testid="loader"]');
    const emptyDivWraper = container.querySelector('[data-testid="emptyDiv"]');
    expect(loaderWrapper).toBeInTheDocument();
    expect(messageWrapper).not.toBeInTheDocument();
    expect(emptyDivWraper).not.toBeInTheDocument();
  });
})