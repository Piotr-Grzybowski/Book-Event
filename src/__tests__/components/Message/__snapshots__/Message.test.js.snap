// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should match snapshot 1`] = `
<div
  className="message"
  data-testid="message"
>
  <p>
    some example message
  </p>
</div>
`;
